pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/filter/UserAuthCheckFilter.java

```diff
@@ -64,6 +68,35 @@ public void handle(Request request, Response response) {
             return;
         }
 
+        boolean canAccess = canAccess(request, ddpAuth);
+        int retries = 3;
+        if (!canAccess) {
+            do {
+                try {
+                    TimeUnit.SECONDS.sleep(1);
+                    log.warn("Retrying checking for access with non-cached permission data");
+                    try {
+                        canAccess = canAccess(request, RouteUtil.computeDDPAuth(request));
+                    } catch (TokenExpiredException e) {
+                        log.error("Found expired token for request", e);
+                        halt(401);
+                    } catch (Exception e) {
+                        log.error("Error while converting token for request", e);
+                        halt(401);
```

Comment:
Found that I needed to return 401 when exceptions occurred to allow existing tests to pass.